src/Engine/EmitMSIL/Replication.cs

```diff
@@ -36,33 +132,35 @@ public static IList ReplicationLogic(IEnumerable<MethodBase> mInfos, IList args,
             //Turn the replication guides into a guide -> List args data structure
             var partialInstructions = Replicator.BuildPartialReplicationInstructions(partialReplicationGuides);
 
-            // TODO: Determine final function endpoint here based on fitting runtime args to function parameters
-            var mi = mInfos.FirstOrDefault();
+            var feps = ConvertMethodsToFEPs(mInfos);
+
+            List<CLRFunctionEndPoint> resolvedFeps;
+            List<ReplicationInstruction> replicationInstructions;
+            ComputeFeps(reducedArgs, feps, partialInstructions, runtimeCore, out resolvedFeps, out replicationInstructions);
+
+            var finalFep = SelectFinalFep(resolvedFeps, reducedArgs);
 
-            // Testing invoking method without replication
             object result;
-            if (mi.IsStatic)
+            if (replicationInstructions.Count == 0)
```

Comment:
do you believe it would be difficult to compute this case at compile time? (no replication)
https://jira.autodesk.com/browse/DYN-5151

Reply:
It should be relatively easy...if we refactor the replication functions a bit...so that we can re-use them at compile time